src/azul/tdr.py

```diff
@@ -224,12 +226,12 @@ def emulate_bundle(self,
             elif link_type == 'supplementary_file_link':
                 # For MVP, only project entities can have associated supplementary files.
                 entity = link['entity']
-                if entity['entity_type'] != 'project' or entity['entity_id'] != links_row['project_id']:
-                    raise ValueError(f'Supplementary file not associated with bundle project: {entity}')
+                require(entity['entity_type'] == 'project' and entity['entity_id'] == links_row['project_id'],
+                        f'Supplmentary files may only be associated with the bundle project: {entity}')
```

Comment:
```suggestion
                        f'Supplementary files may only be associated with the bundle project: {entity}')
```

Reply:
I would split this into two calls to require(), one for each side of the `and`. The the error messages can be more specific about the violation. As in `'… must be associated with the current project ({…}, not {…}).'`